datarobot_batch_scoring/utils.py

```diff
@@ -410,7 +413,8 @@ def investigate_encoding_and_dialect(dataset, sep, ui, fast=False,
     csv.register_dialect('dataset_dialect', dialect)
     #  the csv writer should use the systems newline char
     csv.register_dialect('writer_dialect', dialect,
-                         lineterminator=os.linesep)
+                         lineterminator=os.linesep,
+                         delimiter=(output_delimiter or dialect.delimiter))
```

Comment:
in python 2 we might need to call str() on this. If you give it Unicode the csv parser complains.